addons/project/models/project_task.py

```diff
@@ -794,19 +796,29 @@ def _compute_link_preview_name(self):
                 link_preview_name += f' | {task.project_id.sudo().name}'
             task.link_preview_name = link_preview_name
 
-    @api.depends('is_template', 'parent_id.has_template_ancestor')
+    @api.depends('is_template', 'parent_id.has_template_ancestor', 'is_project_template')
     def _compute_has_template_ancestor(self):
         for task in self:
-            task.has_template_ancestor = task.is_template or (task.parent_id and task.parent_id.has_template_ancestor)
+            task.has_template_ancestor = (
+                task.is_template
+                or (task.parent_id and task.parent_id.has_template_ancestor)
+                or task.is_project_template
+            )
 
     def _search_has_template_ancestor(self, operator, value):
         if operator not in ['=', '!='] or not isinstance(value, bool):
             return NotImplemented
-        template_tasks = self.env['project.task'].with_context(active_test=False).sudo().search([('is_template', '=', True)])
-        domain = [('id', 'child_of', template_tasks.ids)]
+
+        template_tasks = self.env['project.task'].with_context(active_test=False).sudo().search([
+            '|', ('is_template', '=', True), ('is_project_template', '=', True)
```

Comment:
We could probably need an index on `is_template` field of `project.project` model, if we really need to do that.